### PR TITLE
TP10 Documentation and Release Notes updates.

### PR DESCRIPTION
**Pull request by geekspertise**
I've updated the docs and release notes to reflect TP10. This is a work in progress.
Major changes:

- New Operator install process
- New custom resource file
- Old custom resource -> appendix
- Automatic route creation (IOR)
- Updated removal process

Still waiting to see if there are any known issues as a result of testing. 

Can @brian-avery @knrc @tvieira @JStickler @dfennessy and @vramosp please review these changes.

### CODE REVIEW

File: servicemesh-install/topics/appendix-crd-old.adoc
Comment (tvieira) [inline servicemesh-install/topics/appendix-crd-old.adoc]: we may need to change Jaeger version - I'm waiting for confirmation.

File: servicemesh-install/topics/appendix-crd-old.adoc
Comment (tvieira) [inline servicemesh-install/topics/appendix-crd-old.adoc]: we may need to add a new version of Kiali as well - I'm waiting for confirmation.

**Review by dfennessy — APPROVED** [review]

File: servicemesh-install/topics/servicemesh-istio-comparison.adoc
Comment (JStickler) [inline servicemesh-install/topics/servicemesh-istio-comparison.adoc]: Passive voice in this whole section.

File: servicemesh-install/topics/servicemesh-istio-comparison.adoc
Comment (JStickler) [inline servicemesh-install/topics/servicemesh-istio-comparison.adoc]: Can we define what a "catch-all" domain is?

File: servicemesh-install/topics/servicemesh-istio-comparison.adoc
Comment (JStickler) [inline servicemesh-install/topics/servicemesh-istio-comparison.adoc]: by default?  Or upstream just doesn't' support TLS at all?

File: servicemesh-release-notes/modules
Comment (JStickler) [inline servicemesh-release-notes/modules]: Same problem as above, this path won't work for anyone but Elise.

**Review by JStickler — CHANGES_REQUESTED** [review]
I have a lot of questions about the new operator values.  We may not be able to answer them all this go round, but we should try to track down the info if we can.

File: servicemesh-install/topics/appendix-threescale-adapter.adoc
Comment (geekspertise) [reply servicemesh-install/topics/appendix-threescale-adapter.adoc]: I believe @dfennessy is referring to the 3scale Admin Portal URL.

Darren, can you please clarify.

File: servicemesh-install/topics/appendix-threescale-adapter.adoc
Comment (geekspertise) [reply servicemesh-install/topics/appendix-threescale-adapter.adoc]: I believe @dfennessy is referring to the 3scale Admin Portal URL.

Darren, can you please clarify.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: The table columns are "Parameter" "Description" and "Default". The default value is true.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: The table columns are "Parameter" "Description" and "Default". The default value is true.

File: servicemesh-install/topics/install.adoc
Comment (JStickler) [reply servicemesh-install/topics/install.adoc]: Never mind, I just saw Kevin's e-mail comment that "Both are supported in this release, the old one stays around for those who cannot move yet or as a fall back in case we hit some issues."

Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: The custom resource changes in this PR are the new way of managing an Istio installation. However, we are keeping backwards compatibility with the old custom resource for the moment as a fallback. If the operator changes fail, the user still has the option to fall back to using the old custom resource which will trigger the ansible installer.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [inline servicemesh-install/topics/install.adoc]: We're going to revert to deploying the CR in istio-operator for this TP.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: With the new CR, there are several new parameters that we inherited from upstream, many of which enable or disable features and conflict with one another. The model I followed on maistra.io (https://deploy-preview-44--maistra.netlify.com/docs/getting_started/custom-install/) was to start with a working example and then break parameters up by the section that they appear in in the yaml (istio->global, istio->gateways, istio->mixer, istio->pilot, istio->kiali). For each section, I provided an example yaml for that section showing how the parameters would be used as well as a table listing parameters/descriptions/defaults.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: Makes sense to me

File: servicemesh-install/topics/servicemesh-istio-comparison.adoc
Comment (brian-avery) [reply servicemesh-install/topics/servicemesh-istio-comparison.adoc]: There are two types of subdomains considered here. * and *.foo.com. Catch-all refers to *. Open to better terminology.

File: servicemesh-install/topics/servicemesh-istio-comparison.adoc
Comment (brian-avery) [reply servicemesh-install/topics/servicemesh-istio-comparison.adoc]: If the Gateway contains a `tls` section, the OpenShift Route will be configured to support TLS.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: This stems from a recent change upstream. Upstream has decided what they call production and set everything to use that by default. I'm not sure what they used to define that. However, a side effect of this was that a local cluster could not be started for development/test purposes due to the fact that the default configuration without these parameters specifies a significantly larger number of cpus and memory. In terms of product use cases, this is really restricted by the autoscaleMin/autoscaleMax specified below, so if these are configured within reason of a user's environment I don't see an issue with it.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [inline servicemesh-install/topics/install.adoc]: These should be uncommented. After testing Kiali, OAuth is not yet functional.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: These are inherited values from upstream Istio. 
https://istio.io/docs/reference/config/installation-options/#kiali-options

If we want to change terminology, then I think that upstream needs to be updated first.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: Mixer has support for checks and metrics. Checks implement quotas, authorization, authentication, etc. Things that can a user can be denied entry for. This boolean controls what happens when Mixer is unable to contact an adapter implementing the check functionality.  Is the user denied or is the user allowed to pass through without complying?

File: servicemesh-install/topics/install-remove.adoc
Comment (JStickler) [reply servicemesh-install/topics/install-remove.adoc]: @brian-avery, I was thinking that (especially in the remove operation) that you would be referencing files that had been copied to the local server, not pulling from somewhere else?  Because you'd want to be removing the version that you currently had installed, wouldn't you?

File: servicemesh-install/topics/install.adoc
Comment (JStickler) [reply servicemesh-install/topics/install.adoc]: Requests don't go through Mixer though, they go through the proxy to the services.  Isn't that where the pass/deny happens?

Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: Requests do go through Mixer: https://istio.io/docs/concepts/what-is-istio/arch.svg. The pass/deny takes place in Mixer.

Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: Well, request metadata, but the request.

File: servicemesh-install/topics/servicemesh-istio-comparison.adoc
Comment (JStickler) [reply servicemesh-install/topics/servicemesh-istio-comparison.adoc]: I think "wildcard" might be more familiar to people for *.  Open to other ideas though.

File: servicemesh-install/topics/install.adoc
Comment (JStickler) [reply servicemesh-install/topics/install.adoc]: So then we should be saying "whether traffic is allowed to pass through the service mesh"?

File: servicemesh-install/topics/appendix-threescale-adapter.adoc
Comment (dfennessy) [reply servicemesh-install/topics/appendix-threescale-adapter.adoc]: @geekspertise You are correct in both cases :smile:

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: yes

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [reply servicemesh-install/topics/install.adoc]: All of the parameters should be in the examples for each section. Are you seeing any missing ones?

Comment (JStickler) [reply servicemesh-install/topics/install.adoc]: I definitely didn't see mTLS in any of the examples, just in the parameters table.
For Mixer policy, autoscale in disabled in both examples, so there is no sample of autoscaleMin or autoscaleMax in the yaml files.
For Pilot, neither yaml example shows CPU or memory.
There is a parameter table for Tracing, but I don't see it in any of the example yaml files.
In the threescale.yaml file PARAM_THREESCALE_LOG_JSON: is listed twice, I think one should be PARAM_THREESCALE_LOG_LEVEL:

File: servicemesh-install/topics/install-remove.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install-remove.adoc]: Can @knrc and @tvieira possibly weigh in on how we want to handle this?

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: I have fixed the threescale example.

Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: I updated the global example with mTLS and it will be in the update to the PR I push today.

File: servicemesh-install/topics/install-remove.adoc
Comment (tvieira) [reply servicemesh-install/topics/install-remove.adoc]: @geekspertise Do we have a practice of having an appendix to write yaml file examples (i.e. link to appendix C for CRD examples)? I don't like the idea of pointing to Github. We have a moving branch practice and, for the current link, the file already moved to another folder and this link is `404` now. 

Ideally, an appendix with a couple of examples within the document is the best I can think of, so someone can copy and paste within a file and use it as a reference. The additional work with having yaml files in appendix is that we have to test it before pushing changes live, but I'm good adding that to the QE task list.

File: servicemesh-install/topics/install-remove.adoc
Comment (tvieira) [inline servicemesh-install/topics/install-remove.adoc]: BTW, all the links above are wrong now but the operator.

**Review by tvieira — CHANGES_REQUESTED** [review]

File: servicemesh-install/topics/install-remove.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install-remove.adoc]: @tvieira We could place yaml file examples within our documentation set. I would prefer that we had control of the files rather than hosting them elsewhere and serving up bad links, so I agree with you.

File: servicemesh-install/topics/servicemesh-istio-comparison.adoc
Comment (geekspertise) [reply servicemesh-install/topics/servicemesh-istio-comparison.adoc]: I reworked the language in this section.

File: servicemesh-install/topics/install.adoc
Comment (JStickler) [inline servicemesh-install/topics/install.adoc]: re: Rob's comment on https://issues.jboss.org/browse/MAISTRA-366.  I think we should probably revise this to read   "Kiali is installed by default.  If you do not want to install Kiali, change the value to 'enabled: false' to disable the Kiali installation.

Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: I'm inclined to agree. I'll get the language updated for the next PR update.

File: servicemesh-install/topics/appendix-crd-old.adoc
Comment (geekspertise) [reply servicemesh-install/topics/appendix-crd-old.adoc]: Confirmed with @brian-avery that 0.16.2 is correct for TP10.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: I've enabled tracing in the full example.

File: servicemesh-install/topics/install.adoc
Comment (jwendell) [inline servicemesh-install/topics/install.adoc]: I believe this section is not in the right place. We are instructing the user to update a thing that is not even installed (We're going to install only in the next section). I believe we should move that to any place below installation section

**Review by jwendell — CHANGES_REQUESTED** [review]

File: servicemesh-install/topics/install-remove.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install-remove.adoc]: The yaml files were condensed down to one servicemesh-operator.yaml.

File: servicemesh-install/topics/install-remove.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install-remove.adoc]: The yamls have been trimmed down into one servicemesh-operator.yaml link that works.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: This known issue is in the update I pushed to this PR.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: This has been resolved and there is only one servicemesh-operator.yaml file now.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: In order for what is installed to work, the Mixer policy enforcement status must be enabled. I think it makes more sense to do it _before_ the install, otherwise, people will forget to do it and report issues.

File: servicemesh-install/topics/install-remove.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install-remove.adoc]: This is corrected.

File: servicemesh-install/topics/appendix-crd-old.adoc
Comment (geekspertise) [reply servicemesh-install/topics/appendix-crd-old.adoc]: 0.16.2 is shipping with TP10.

**Comment by jwendell** [comment]
in the section "Add default destination rules"
If you did not enable mutual TLS:
replace the two commands with this single one:
`oc apply -n myproject -f https://raw.githubusercontent.com/istio/istio/release-1.1/samples/bookinfo/networking/destination-rule-all.yaml`
and, in the following item (If you enabled mutual TLS:) replace the two commands with this single one:
`oc apply -n myproject -f https://raw.githubusercontent.com/istio/istio/release-1.1/samples/bookinfo/networking/destination-rule-all-mtls.yaml`

File: servicemesh-install/topics/appendix-crd-old.adoc
Comment (tvieira) [inline servicemesh-install/topics/appendix-crd-old.adoc]: @geekspertise if the user decides to use the old resource definition, we need to make sure that that the user needs to apply the resource to `-n istio-operator` instead of `-n istio-system` otherwise it will fail.

**Review by tvieira — CHANGES_REQUESTED** [review]

File: servicemesh-install/topics/install-post.adoc
Comment (brian-avery) [inline servicemesh-install/topics/install-post.adoc]: We might want a comment here saying that istio-installation should be the name of the resource

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [inline servicemesh-install/topics/install.adoc]: is disabled by default. You must enable it to run policy tasks.

File: servicemesh-install/topics/appendix-crd-old.adoc
Comment (geekspertise) [reply servicemesh-install/topics/appendix-crd-old.adoc]: I added this note:

If you are using the previous version of the custom resource definition (linked), apply the resource to istio-operator instead of istio-system or the install will fail.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [inline servicemesh-install/topics/install.adoc]: @tvieira Here is the note I added.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [inline servicemesh-install/topics/install.adoc]: deployment_type no longer exists

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [inline servicemesh-install/topics/install.adoc]: same here. deployment_type no longer exists

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: Fixed.

File: servicemesh-install/topics/install-post.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install-post.adoc]: I added a note stating: "The name of the resource is istio-installation."

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: Fixed.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: Fixed.

File: servicemesh-install/topics/install.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: Fixed.

File: servicemesh-install/topics/install-post.adoc
Comment (geekspertise) [reply servicemesh-install/topics/install-post.adoc]: Added a note.

File: servicemesh-install/topics/install.adoc
Comment (brian-avery) [inline servicemesh-install/topics/install.adoc]: I think mutual TLS is better than Mutual Transport Layer Security

Comment (JStickler) [reply servicemesh-install/topics/install.adoc]: The problem with acronyms is that there are usually multiple possible definitions.  Acronym Finder has seven for MTLS.  Spelling it out helps anyone who doesn't know what the acronym stands for (especially folks who have English as a second language.)

**Comment by geekspertise** [comment]
@vikram-redhat and @kalexand-rh Can we merge this PR? It's been reviewed by @JStickler and @knrc said we're good to merge as TP10 is live.

File: servicemesh-install/topics/install.adoc
Comment (kalexand-rh) [inline servicemesh-install/topics/install.adoc]: @geekspertise, did you mean to link to 3.11 instead of 3.1?

Comment (geekspertise) [reply servicemesh-install/topics/install.adoc]: @kalexand-rh I updated the link and updated the PR.

**Comment by kalexand-rh** [comment]
@geekspertise, thanks! It scans ok, and I'll merge it after the build passes.

**Pull request merged**
